### PR TITLE
Migrate `server` module tests to JUnit 5, part 6

### PR DESCRIPTION
**Pull request by dmitrykuzmin**
This PR continues migration of the `server` module to the new JUnit 5 test structure.

The following packages were migrated:
1. `projection`
2. `reflect`
3. `rejection`
4. `route`

### CODE REVIEW

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.

File: server/src/test/java/io/spine/server/projection/ProjectionRepositoryTest.java [hunk 1]
```diff
@@ -171,111 +175,121 @@ public void setUp() {
...
-        projection = repository().findOrCreate(projectId);
-        List<Task> addedTasks = projection.getState()
-                                          .getTaskList();
-        assertTrue(projection.isDeleted());
-
-        // Check that the projection was not re-created before dispatching.
-        assertFalse(addedTasks.isEmpty());
-    }
+        @Test
+        @DisplayName("event to archived projection")
```
Comment (armiol) [inline server/src/test/java/io/spine/server/projection/ProjectionRepositoryTest.java]: I think events are NOT dispatched to archived and deleted instances.

Please check. That makes difference to the display name of these tests.

File: server/src/test/java/io/spine/server/projection/ProjectionRepositoryTest.java [hunk 2]
```diff
@@ -342,7 +365,8 @@ public void do_not_create_record_if_entity_is_not_updated() {
      * Beam-based catch-up are exposed.
      */
     @Test
-    public void expose_read_and_write_methods_for_last_handled_event_timestamp() {
+    @DisplayName("expose read and write methods for last handled event timestamp")
```
Comment (armiol) [inline server/src/test/java/io/spine/server/projection/ProjectionRepositoryTest.java]: `for the timestamp of the last handled event`

File: server/src/test/java/io/spine/server/projection/ProjectionRepositoryTest.java [hunk 3]
```diff
@@ -352,30 +376,39 @@ public void expose_read_and_write_methods_for_last_handled_event_timestamp() {
...
+        @Test
+        @DisplayName("bounded context")
+        void boundedContext() {
+            assertNotNull(repository().boundedContext());
+        }
     }
 
     @Test
-    public void throw_exception_on_attempt_to_register_in_bc_with_no_messages_handled() {
+    @DisplayName("throw exception on attempt to register in BC with no messages handled")
```
Comment (armiol) [inline server/src/test/java/io/spine/server/projection/ProjectionRepositoryTest.java]: In fact there is a mistake in the method name. Projections subscribe to messages. We use `to handle` in relation to commands only.

Let's rename the method and fix the display name accordingly.

File: server/src/test/java/io/spine/server/rejection/RejectionBusTest.java [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+            rejectionBus.unregister(subscriberOne);
+
+            // Check that the 2nd subscriber with the same rejection subscriber method remains
+            // after the 1st subscriber unregisters.
+            final Collection<RejectionDispatcher<?>> subscribers =
+                    rejectionBus.getDispatchers(rejectionClass);
+            assertFalse(subscribers.contains(subscriberOne));
+            assertTrue(subscribers.contains(subscriberTwo));
+
+            // Check that after 2nd subscriber us unregisters he's no longer in
```
Comment (armiol) [inline server/src/test/java/io/spine/server/rejection/RejectionBusTest.java]: `Check that after 2nd subscriber is unregistered, he's no longer in.`

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.

**Comment by armiol** [comment]
@dmitrykuzmin LGTM.

**Pull request merged**